include/onnxruntime/core/session/environment.h

```diff
@@ -5,11 +5,13 @@
 
 #include <atomic>
 #include <memory>
+#include <unordered_set>
 #include "core/common/common.h"
 #include "core/common/status.h"
 #include "core/platform/threadpool.h"
 #include "core/common/logging/logging.h"
 #include "core/framework/allocator.h"
+#include "core/session/onnxruntime_c_api_ep.h"
 
```

Comment:
Can we forward declare the types and avoid public header inclusion in the header?